packages/dds/tree/src/shared-tree-core/sharedTreeCore.ts

```diff
@@ -217,19 +226,27 @@ export class SharedTreeCore<TEditor extends ChangeFamilyEditor, TChange> extends
 	 * @param revision - The revision to associate with the change.
 	 * Defaults to a new, randomly generated, revision if not provided.
 	 */
-	protected applyChange(change: TChange, revision?: RevisionTag): void {
+	protected applyChange(change: TChange, type?: GraphCommitType, revision?: RevisionTag): void {
 		const commit = {
 			change,
 			revision: revision ?? mintRevisionTag(),
 			sessionId: this.editManager.localSessionId,
+			type,
 		};
+		const changeDelta = this.changeFamily.intoDelta(change);
```

Comment:
It looks like we could just use the return value of `this.editManager.addLocalChange` instead of recomputing this separately. They should be exatly the same unless I'm missing something. This duplicated computation issue predates your changes, maybe Noah can double-check if there was a reason for it.

Reply:
I think this is being computed separately because the capture needs to happen before the change is applied. I also use it in my call to capture below.

Reply:
The capture needs to happen before the changes is applied to the forest. I don't think `this.editManager.addLocalChange` applies the change to the forest (the `this.changeEvents.emit` calls do, unless I'm mistaken). Or are you saying the forest is not our only concern here?